tests/components/guardian/test_config_flow.py

```diff
@@ -40,6 +44,18 @@ async def test_connect_error(hass):
         assert result["errors"] == {CONF_IP_ADDRESS: "cannot_connect"}
 
 
+async def test_get_pin_from_discovery_hostname():
+    """Test getting a device PIN from the zeroconf-discovered hostname."""
+    pin = async_get_pin_from_discovery_hostname("GVC1-3456.local.")
+    assert pin == "3456"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5596821)